packages/vue-app/template/index.js

```diff
@@ -66,6 +67,14 @@ const defaultTransition = <%=
 %><%= isTest ? '// eslint-disable-line' : '' %>
 <% } %>
 
+const { assign, create } = Object
+const originalRegisterModule = Vuex.Store.prototype.registerModule
+const baseStoreOptions = { preserveState: process.client }
+
+function registerModule (path, rawModule, options) {
+  originalRegisterModule.call(this, path, rawModule, assign(create(baseStoreOptions), options))
```

Comment:
- Any reason not using { preserveState, ...options } ?
- Currently registerModule isn't returning anything but maybe better if we preserve it's return value?

Reply:
I thought `...` produced some horrible transpiled code? Also with create() won't we be further limiting allocation of at least one property? Good call about adding a `return`.

Reply:
No idea why the build failed -- will run locally again.

Reply:
@galvez We are using spread syntax several places at least in `client.js` and babel helpers (which are not that heavy) already injected once. Also babel rules are different for modern. [example transpile](https://babeljs.io/en/repl#?browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=MYewdgzgLgBAhjAvDA3jAZiEAuGUBOArgKYA0MAdFQEZz4wC-QA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Cenv&prettier=false&targets=&version=7.11.1&externalPlugins=)

Reply:
@pi0 updated! Also reworded the pR title so you can classify this as a general enhancement.

It does seem like the memory leak is valid, since that function allocation is happening repeatedly.